LoginDemo/LearningSelenium/FactoryDesign/FactoryMethodDesign/IFMProduct.cs

```diff
@@ -0,0 +1,127 @@
+﻿using OpenQA.Selenium;
+
+namespace LearningSelenium.FactoryDesign.FactoryMethodDesign
+{
+    public interface IFMProduct
+    {
+
+        void PerformClick();
+    }
+
+    //concrete class
+    public class FMDLogin : IFMProduct
+    {
+        private readonly WebDriver driver;
+        private readonly ReadConfig config;
+        private By swagUserName = By.XPath("//input[contains(@id,'user-name')]");
+        private By swagPassword = By.XPath("//input[contains(@id,'password')]");
+        private By swagLogButton = By.XPath("//input[contains(@id,'login-button')]");
+
+        public FMDLogin(WebDriver driver, ReadConfig config)
+        {
+            this.driver = driver;
+            this.config = config;
+        }
+
+        public void EnterSwagUserName(string userName)
+        {
+            driver.FindElement(swagUserName).SendKeys(userName);
+        }
+
+        public void EnterSwagPassword(string password)
+        {
+            driver.FindElement(swagPassword).SendKeys(password);
+        }
+
+        public void PerformClick()
```

Comment:
According to the name of this method it has to be only: driver.FindElement(swagLogButton).Click();

Reply:
enter details are created as a separate method